lib/libkafka_asio/primitives.h

```diff
@@ -30,6 +30,11 @@ typedef boost::uint_t<8>::exact Byte;
 typedef std::string String;
 typedef boost::shared_ptr<std::vector<Byte> > Bytes;
 
+// Default Properties (empty)
```

Comment:
Please move this back to `topic_partition_block.h` (and include it in `topic_partition_map.h` or vice-versa).
The `primitives.h` contains only those types, defined in the Kafka wire protocol here:
https://cwiki.apache.org/confluence/display/KAFKA/A+Guide+To+The+Kafka+Protocol#AGuideToTheKafkaProtocol-ProtocolPrimitiveTypes